base/shell/explorer/startctxmnu.cpp

```diff
@@ -49,26 +55,26 @@ class CStartMenuBtnCtxMenu :
                 hRet = m_Inner->QueryContextMenu(
                     hPopup,
                     0,
-                    ID_SHELL_CMD_FIRST,
-                    ID_SHELL_CMD_LAST,
+                    idCmdFirst,
+                    idCmdLast,
                     CMF_VERBSONLY);
 
                 if (SUCCEEDED(hRet))
                 {
                     return hRet;
                 }
-
-                DestroyMenu(hPopup);
```

Comment:
Is this now handled by the caller of `CreateContextMenuFromShellFolderPidl()`?

Reply:
It has never been valid to destroy the menu passed to you in QueryContextMenu. The call above is extremely likely to succeed so nobody noticed.